homeassistant/components/motioneye/camera.py

```diff
@@ -162,36 +153,27 @@ def _set_mjpeg_camera_state_for_camera(self, camera: dict[str, Any]) -> None:
         if self._authentication == HTTP_BASIC_AUTHENTICATION:
             self._auth = aiohttp.BasicAuth(self._username, password=self._password)
 
-    @property
-    def unique_id(self) -> str:
-        """Return a unique id for this instance."""
-        return self._unique_id
-
-    @classmethod
-    def _is_acceptable_streaming_camera(cls, camera: dict[str, Any] | None) -> bool:
+    def _is_acceptable_streaming_camera(self) -> bool:
         """Determine if a camera is streaming/usable."""
-        return is_acceptable_camera(camera) and MotionEyeClient.is_camera_streaming(
-            camera
-        )
+        return is_acceptable_camera(
+            self._camera
+        ) and MotionEyeClient.is_camera_streaming(self._camera)
 
     @property
     def available(self) -> bool:
         """Return if entity is available."""
-        return self._available
+        return super().available and self._is_acceptable_streaming_camera()
 
     @callback
     def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
-        available = False
-        if self.coordinator.last_update_success:
-            camera = get_camera_from_cameras(self._camera_id, self.coordinator.data)
-            if self._is_acceptable_streaming_camera(camera):
-                assert camera
-                self._set_mjpeg_camera_state_for_camera(camera)
-                self._motion_detection_enabled = camera.get(KEY_MOTION_DETECTION, False)
-                available = True
-        self._available = available
         super()._handle_coordinator_update()
```

Comment:
Isn't calling `super()` already triggering an `async_write_ha_state()`? In that case, better move the `if` statement before this one and remove the write_ha_state statement .

Reply:
Yes, but the `super()` is also updating `self._camera` which is used in the if statement. So I can do as you suggest, but then I need to duplicate the `self._camera =` here (and/or in other entities), so not sure it's worth it?

Reply:
Hmm. 

I still think we should try to avoid calling write_ha_state twice. Can we break the super function out in "do_work(); write_ha_state()" and we can just call do_work() there ?